core/src/main/java/com/yahoo/ycsb/workloads/RestWorkload.java

```diff
@@ -0,0 +1,300 @@
+
+  /**
+   * The name of the property for the proportion of transactions that are
+   * delete.
+   */
+  public static final String DELETE_PROPORTION_PROPERTY = "deleteproportion";
+
+  /**
+   * The default proportion of transactions that are delete.
+   */
+  public static final String DELETE_PROPORTION_PROPERTY_DEFAULT = "0.00";
+
+  /**
+   * The name of the property for the file that holds the field length size for insert operations.
+   */
+  public static final String FIELD_LENGTH_DISTRIBUTION_FILE_PROPERTY = "fieldlengthdistfile";
+  
+  /**
+   * The default file name that holds the field length size for insert operations.
+   */
+  public static final String FIELD_LENGTH_DISTRIBUTION_FILE_PROPERTY_DEFAULT = "fieldLengthDistFile.txt";
+
+  /**
+   * In web services even though the CRUD operations follow the same request
+   * distribution, they have different traces and distribution parameter
+   * values. Hence configuring the parameters of these operations separately
+   * makes the benchmark more flexible and capable of generating better
+   * realistic workloads.
+   */
+  // Read related properties.
+  private static final String READ_TRACE_FILE = "url.trace.read";
+  private static final String READ_TRACE_FILE_DEFAULT = "readtrace.txt";
+  private static final String READ_ZIPFIAN_CONSTANT = "readzipfconstant";
+  private static final String READ_ZIPFIAN_CONSTANT_DEAFULT = "0.99";
+  private static final String READ_RECORD_COUNT_PROPERTY = "readrecordcount";
+  // Insert related properties.
+  private static final String INSERT_TRACE_FILE = "url.trace.insert";
+  private static final String INSERT_TRACE_FILE_DEFAULT = "inserttrace.txt";
+  private static final String INSERT_ZIPFIAN_CONSTANT = "insertzipfconstant";
+  private static final String INSERT_ZIPFIAN_CONSTANT_DEAFULT = "0.99";
+  private static final String INSERT_SIZE_ZIPFIAN_CONSTANT = "insertsizezipfconstant";
+  private static final String INSERT_SIZE_ZIPFIAN_CONSTANT_DEAFULT = "0.99";
+  private static final String INSERT_RECORD_COUNT_PROPERTY = "insertrecordcount";
+  // Delete related properties.
+  private static final String DELETE_TRACE_FILE = "url.trace.delete";
+  private static final String DELETE_TRACE_FILE_DEFAULT = "deletetrace.txt";
+  private static final String DELETE_ZIPFIAN_CONSTANT = "deletezipfconstant";
+  private static final String DELETE_ZIPFIAN_CONSTANT_DEAFULT = "0.99";
+  private static final String DELETE_RECORD_COUNT_PROPERTY = "deleterecordcount";
+  // Delete related properties.
+  private static final String UPDATE_TRACE_FILE = "url.trace.update";
+  private static final String UPDATE_TRACE_FILE_DEFAULT = "updatetrace.txt";
+  private static final String UPDATE_ZIPFIAN_CONSTANT = "updatezipfconstant";
+  private static final String UPDATE_ZIPFIAN_CONSTANT_DEAFULT = "0.99";
+  private static final String UPDATE_RECORD_COUNT_PROPERTY = "updaterecordcount";
+
+  private static Map<Integer, String> readUrlMap;

```

Comment:
Good point! They will be accessed by different threads but only for reading. But yes it makes sense to not keep them static as a single YCSB instance can be used for multiple benchmarking with different traces. Change incorporated.